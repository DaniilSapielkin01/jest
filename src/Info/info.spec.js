import React from 'react';
import { describe } from 'yargs';
import Info from './info';

const setUp = (props) => shallow(<Info />);

const componentDidMountSpy = jest.spyOn(Info.prototype, 'componentDidMount');
const componentDidUpdateSpy = jest.spyOn(Info.prototype,'componentDidUpdate');
const componentWillUnmountSpy = jest.spyOn(Info.prototype,'componentWillUnmount');


describe("Info component",()=>{
    let component;
    beforeEach(()=> { 
        jest.spyOn(window, 'addEventListener');
        jest.spyOn(window, 'removeEventListener');
        component = setUp();

    });

    afterEach(()=>{
        window.addEventListener.mockRestore();
        window.removeEventListener.mockRestore();
    }); 

    it("should render info",()=> { 
        expect(component).toMatchSnapshot(); 
    });

    describe('Lifecycle methods', ()=>{
        it("should call componentDidMount one ",()=> { 
         expect(componentDidMountSpy).toHaveBeenCalledTimes(1);
        });

        it("should not call componentWillUnmount when componet just mounted ",()=> { 
            expect(componentDidMountSpy).toHaveBeenCalledTimes(1);
            expect(componentWillUnmountSpy).toHaveBeenCalledTimes(0);
           });

           it('should componentDidUpdate',() => { 
            component.setProps();
            expect(componentDidUpdateSpy).toHaveBeenCalled()
           });

           it('should componentWillUnmount',() => { 
            component.unmount();

            expect(componentWillUnmountSpy).toHaveBeenCalledTimes(1)
           });

           describe('Component handlers', () => { 
            it('should call addEventListener when component mounted',()=> { 
                expect(window.addEventListener).toHaveBeenCalledTimes(1);
            });

            it('should call handleChangeTitle in componentDidUpdate',()=> { 
                const instance = setUp().instance();
                instance.handleChangeTitle = jest.fn();
                expect(instance.handleChangeTitle).toHaveBeenCalled();

            });

            it('should call removeEventListener when component unmounted',()=> { 
                component.unmount();

               expect(window.addEventListener).toHaveBeenCalledTimes(1);
            });
            
            it("should call handleWidth during window resize",()=> {
                expect(component.state().width).toBe(0);
                //вызываем рисайз
                global.dispatchEvent(new Event('resize'));
                expect(component.state().width).toBe(window.innerWidth)

            });


        });
    });

});
