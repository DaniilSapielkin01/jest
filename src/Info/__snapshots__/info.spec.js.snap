// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info component should render info 1`] = `
<h1>
  0
</h1>
`;
